src/core/components/SearchChatHistory.tsx

```diff
@@ -1,53 +1,68 @@
 import { useState, useRef, useEffect } from 'react';
-import { Search } from 'lucide-react';
 import styles from './SearchChatHistory.module.css';
-import { ConversationHistory } from '../context/types';
 import ModalWrapper from './ModalWrapper';
 import SearchModalBody from './SearchModalBody';
+import { useAppContext } from '../context/useAppContext';
+import { SearchChatButton } from '../assets/SearchChatButton';
+import { ConversationHistory } from '../context/types';
 
 interface SearchModalProps {
-  conversations: ConversationHistory[];
-  onConversationSelect: (conversation: ConversationHistory) => void;
+  isChatHistoryOpen: boolean;
+  setIsChatHistoryOpen: (i: boolean) => void;
+  setIsMobileSideBarOpen: (i: boolean) => void;
 }
 
 const SearchChatHistory = ({
-  conversations,
-  onConversationSelect,
+  isChatHistoryOpen,
+  setIsChatHistoryOpen,
+  setIsMobileSideBarOpen,
 }: SearchModalProps) => {
-  const [isOpen, setIsOpen] = useState(false);
+  const { conversations, hasConversations, loadConversation } = useAppContext();
   const [searchTerm, setSearchTerm] = useState('');
   const modalRef = useRef<HTMLDivElement>(null);
   const inputRef = useRef<HTMLInputElement>(null);
 
   useEffect(() => {
-    if (isOpen) {
+    if (isChatHistoryOpen) {
       // Focus input when modal opens
       inputRef.current?.focus();
+      //  mobile sidebar is closed by default and that contains the component that mounts the search modal,
+      //  but if a user has opened chat history search from a larger screen, then we should update that so search modal
+      //  stays open on screen resize
+      setIsMobileSideBarOpen(true);
+    }
```

Comment:
Demo without setting this - it works once the user sets the mobile sidebar open, but not initially.

Not a super likely userflow, but at least its covered


https://github.com/user-attachments/assets/0b1f67df-f001-47df-9168-c31ff6ba28fd